puzzles/puzzles.go

```diff
@@ -56,19 +66,135 @@ func CreatePuzzlesFromGame(conf *config.Config, g *game.Game) ([]*pb.PuzzleCreat
 	return puzzles, nil
 }
 
-func EquityPuzzle(moves []*move.Move) (bool, pb.PuzzleTag) {
+func EquityPuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
 	return len(moves) >= 2 && moves[0].Equity() > moves[1].Equity()+10, pb.PuzzleTag_EQUITY
 }
 
-func OnlyBingoPuzzle(moves []*move.Move) (bool, pb.PuzzleTag) {
+func BingoPuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
+	m := moves[0]
+	return moveIsBingo(m), pb.PuzzleTag_BINGO
+}
+
+func OnlyBingoPuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
 	tag := pb.PuzzleTag_ONLY_BINGO
-	if len(moves) == 0 || moves[0].TilesPlayed() != 7 {
+	if len(moves) == 0 || !moveIsBingo(moves[0]) {
 		return false, tag
 	}
-	for _, mv := range moves[1:] {
-		if mv.TilesPlayed() == 7 {
+	for _, m := range moves[1:] {
+		if moveIsBingo(m) && m.Action() == move.MoveTypePlay {
 			return false, tag
 		}
 	}
 	return true, tag
 }
+
+func BlankBingoPuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
+	m := moves[0]
+	return moveIsBingo(m) && moveContainsBlank(m), pb.PuzzleTag_BLANK_BINGO
+}
+
+func NonBingoPuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
+	return !moveIsBingo(moves[0]), pb.PuzzleTag_NON_BINGO
+}
+
+// XXX: Must be expanded to other languages
+func PowerTilePuzzle(g *game.Game, moves []*move.Move) (bool, pb.PuzzleTag) {
+	return moveContainsLetter(moves[0], "XJZQ"), pb.PuzzleTag_POWER_TILE
```

Comment:
how about we look at the game's alphabet, and figure out from there if it has a letter that is worth `X` or more points (6, perhaps? 8?)